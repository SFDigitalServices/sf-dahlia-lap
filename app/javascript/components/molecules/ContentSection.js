import React from 'react'
import classNames from 'classnames'

const ContentSection = ({ title, description, children }) => (
  <div className='app-inner padding-bottom--half padding-top--half'>
    <ContentSection.Header title={title} description={description} />
    {children}
  </div>
)

ContentSection.Header = ({ title, description }) => (
  <>
    {title && <h2 className='app-card_h2'>{title}</h2>}
    {description && <p className='form-note max-width margin-bottom--3halves'>{description}</p>}
  </>
)

ContentSection.SubHeader = ({ title, description }) => (
  <>
    {title && <h3 className='app-card_h3 t-sans'>{title}</h3>}
    {description && <p className='form-note max-width margin-bottom--3halves'>{description}</p>}
  </>
)

ContentSection.Content = ({ children, borderBottom, marginTop }) => {
  const divClassName = classNames(
    {
      'border-bottom': borderBottom,
      'margin-top': marginTop
    }
  )
  return (
    <div className={divClassName} style={{ marginBottom: '1.75rem' }}>
      {children}
    </div>
  )
}

ContentSection.Sub = ({ title, description, borderBottom = false, children }) => (
  <>
    <ContentSection.SubHeader title={title} description={description} />
    <ContentSection.Content borderBottom={borderBottom}>
      {children}
    </ContentSection.Content>
  </>
)

export default ContentSection
