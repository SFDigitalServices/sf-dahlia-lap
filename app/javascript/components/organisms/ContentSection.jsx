import React from 'react'
import FormGridRow from '../molecules/FormGridRow'

const ContentSection = ({ title, subtitle, label, id, name, placeholder, describeId, note, error }) => {
  return (
    <div className="content-section">
      <div className="app-inner header-wide">
        <h3 className="app-card_h2">{title}</h3>
        <p className="form-note max-width">{subtitle}</p>
      </div>
      <div className="app-inner inset-wide border-bottom">
        <FormGridRow label={label} id={id} name={name} placeholder={placeholder} describeId={describeId} note={note} error={error} />
      </div>
    </div>
  )
}

export default ContentSection
