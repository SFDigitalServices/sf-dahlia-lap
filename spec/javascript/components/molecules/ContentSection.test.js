import React from 'react'

import renderer from 'react-test-renderer'

import ContentSection from 'components/molecules/ContentSection'

describe('ContentSection', () => {
  test('should render correctly', () => {
    const wrapper = renderer.create(
      <ContentSection title='main title' description='main desc'>
        <ContentSection.Sub title='sub title' description='sub desc'>
          content
        </ContentSection.Sub>
      </ContentSection>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
