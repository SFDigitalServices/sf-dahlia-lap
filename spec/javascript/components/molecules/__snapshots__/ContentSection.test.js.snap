// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentSection should render correctly 1`] = `
<div
  className="app-inner padding-bottom--half padding-top--half"
>
  <h2
    className="app-card_h2"
  >
    main title
  </h2>
  <p
    className="form-note max-width margin-bottom--3halves"
  >
    main desc
  </p>
  <h3
    className="app-card_h3 t-sans"
  >
    sub title
  </h3>
  <p
    className="form-note max-width margin-bottom--3halves"
  >
    sub desc
  </p>
  <div
    className=""
    style={
      Object {
        "marginBottom": "1.75rem",
      }
    }
  >
    content
  </div>
</div>
`;
